clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java

```diff
@@ -369,6 +369,20 @@ public String[] listRoleNames() throws NoSuchMetalakeException {
     return getMetalake().listRoleNames();
   }
 
+  /**
+   * Lists the role names associated with a metadata object.
+   *
+   * @param object The object associated with the role.
+   * @return The role name list.
+   * @throws NoSuchMetalakeException If the Metalake with the given name does not exist.
+   * @throws NoSuchMetadataObjectException If the Metadata object with the given name does not
+   *     exist.
+   */
+  public String[] listRoleNamesByObject(MetadataObject object)
+      throws NoSuchMetalakeException, NoSuchMetadataObjectException {
+    return getMetalake().listRoleNamesByObject(object);
```

Comment:
It is better to design the APIs like tags, list roles from different entity object.

Reply:
Different from Tag API, role is related to authorization. Only we enable authorization, entity can list roles. So I prefer making a dedicated API to do this. WDYT?

Reply:
Do you honor authorization here? I don't see you honor the authorization here even you put the code here.

Reply:
OK, I can refer to the Tag to modify here.

Reply:
Think twice. Different from tag, tag don't support `metalake` type. role supports`metalake`.
 If we add `listObjectRoleNames` API. It will make users confusing.
What's the difference between `listObjectRoleNames` and `listRoleNames` for a metalake.

Reply:
We don't want to expose `GravitinoMetalake`. If we want to list roles for `GravitinoMetalake`, it's hard.

Reply:
How about using different method name like "listBindingRoleNames"?

Reply:
You can add more descriptions to tell about the difference.

Reply:
OK, I modified this code as your suggestion.